package NYT.NYTree.NProto;

import "yt/core/rpc/rpc.proto";

////////////////////////////////////////////////////////////////////////////////

message TYPathHeaderExt
{
    extend NRpc.NProto.TRequestHeader
    {
        optional TYPathHeaderExt ypath_header_ext = 103;
    }

    optional bool mutating = 1 [default = false];
}

////////////////////////////////////////////////////////////////////////////////

message TAttributeFilter
{
    required int32 mode = 1;
    repeated string keys = 2;
}

message TReqGet
{
    optional TAttributeFilter attribute_filter = 1;
    optional int64 max_size = 2;
    optional bool ignore_opaque = 3;
}

message TRspGet
{
    required bytes value = 1;
}

////////////////////////////////////////////////////////////////////////////////

message TReqGetKey
{ }

message TRspGetKey
{
    required bytes value = 1;
}

////////////////////////////////////////////////////////////////////////////////

message TReqSet
{
    required bytes value = 1;
}

message TRspSet
{ }

////////////////////////////////////////////////////////////////////////////////

message TReqRemove
{
    optional bool recursive = 1 [default = true];
    optional bool force = 2 [default = false];
}

message TRspRemove
{ }

////////////////////////////////////////////////////////////////////////////////

message TReqList
{
    optional TAttributeFilter attribute_filter = 1;
    optional int64 max_size = 2;
}

message TRspList
{
    required bytes keys = 1;
}

////////////////////////////////////////////////////////////////////////////////

message TReqExists
{ }

message TRspExists
{
    required bool value = 1;
}

////////////////////////////////////////////////////////////////////////////////

